### PR TITLE
Monitor Query SDK README and samples updates

### PR DESCRIPTION
**Pull request by scottaddie**
**Summary of changes**
- Change occurrences of `LogsClient` and `MetricsClient` to `LogsQueryClient` and `MetricsQueryClient`, respectively
- Use `var` where the explicit type is obvious
- Reduce horizontal scrolling for code samples
- Conduct an edit pass on README file

### CODE REVIEW

**Review by pakrym — APPROVED** [review]
❤️

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 1]
```diff
@@ -1,44 +1,46 @@
...
+### Authenticate the client
 
-In order to interact with the Azure Monitor service, you will need to create an instance of a [TokenCredential](https://docs.microsoft.com/dotnet/api/azure.core.tokencredential?view=azure-dotnet) class and pass it to the constructor of your `LogsClient` or `MetricsClient` class.
+To interact with the Azure Monitor service, create an instance of a [TokenCredential](https://docs.microsoft.com/dotnet/api/azure.core.tokencredential?view=azure-dotnet) class. Pass it to the constructor of your `LogsQueryClient` or `MetricsQueryClient` class.
 
 ## Key concepts
 
-- `LogsClient` - Client that provides methods to query logs from Azure Monitor Logs.
-- `MetricsClient` - Client that provides methods to query metrics from Azure Monitor Metrics.
+* `LogsQueryClient`&mdash;Client that provides methods to query logs from Azure Monitor Logs.
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:35]: `&mdash;` might not work in nuget readmes as it doesn't support embedded HTML

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 2]
```diff
@@ -51,23 +53,26 @@ We guarantee that all client instance methods are thread-safe and independent of
 
 ## Examples
 
-- [Query logs](#query-logs)
-- [Query logs as model](#query-logs-as-model)
-- [Query logs as primitive](#query-logs-as-primitive)
-- [Batch query](#batch-query)
-- [Query dynamic table](#query-dynamic-table)
-- [Increase query timeout](#increase-query-timeout)
+* [Query logs](#query-logs)
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:56]: We use `-` in other readmes. Is `*` better in some way?

Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/README.md:56]: We use `-` in other readmes. Is `*` better in some way?

Comment (scottaddie) [reply sdk/monitor/Azure.Monitor.Query/README.md:56]: Let's standardize on `-` then. I'll update it.

**Pull request merged**